src/VisualStudio/CSharp/Impl/ChangeSignature/CSharpChangeSignatureViewModelFactoryService.cs

```diff
@@ -44,7 +44,7 @@ public override SymbolDisplayPart[] GeneratePreviewDisplayParts(AddedParameterVi
             return parts.ToArray();
         }
 
-        public override bool IsTypeNameValid(string typeName) => !SyntaxFactory.ParseTypeName(typeName).ContainsDiagnostics;
+        public override bool IsTypeNameValid(string typeName, ParseOptions parseOptions) => !SyntaxFactory.ParseTypeName(typeName, options: parseOptions).ContainsDiagnostics;
```

Comment:
still confsued about this.  not sure why parseoptions are necessary to ask if a tyep name is valid.

Reply:
note: imo, what we shoudl be doing here is removing this options, and just passing in Latest to ParseTypeName.  that way we're effectively saying "are there any diagnostics *unrelated* to "you're using teh wrong langversion".

Reply:
`delegate*<void>` is _syntactically_ invalid if the language version is not `Preview`. We don't have this issue with any existing types today (even `nint`) because they can't be determined to be invalid from syntax alone. I'm certainly fine with passing `Preview` as the default langversion however, as we'll appropriately error about langversion when it's actually added.

Reply:
Agreed, seems best to just pass in latest. What's funky about trying to do a check here is it's still not necessarily valid everywhere _else_ the actual fix might apply to, so it's not really protecting the user from much.

Reply:
Now passing latest.